api/turing/api/alerts_api.go

```diff
@@ -10,12 +10,12 @@ import (
 )
 
 type AlertsController struct {
-	*baseController
+	*BaseController
 }
 
 var ErrAlertDisabled = errors.New("alert is disabled in turing-api")
 
-func (c *AlertsController) CreateAlert(r *http.Request, vars map[string]string, body interface{}) *Response {
+func (c AlertsController) CreateAlert(r *http.Request, vars RequestVars, body interface{}) *Response {
```

Comment:
1. It's much easier: I did it for the consistency with other controllers (for example, [outers_api.go](https://github.com/gojek/turing/blob/main/api/turing/api/routers_api.go)). Generally, I can't see any reasons of why it should be a pointer receiver (i.e, mutability is not required, performance is not a concern, etc.)
2. Because BaseController is also defined with pointer receivers (I missed that) and initially I added `Routes() []Route` [method](https://github.com/romanwozniak/turing/blob/a05dc201c839a036e3c2f05c24939fe22d46d7d7/api/turing/api/base_controller.go#L12) into it to implement `Controller` interface and stub methods of that interface for all the controllers that have BaseController embedded. Since `BaseController` is defined with pointer receivers, it should be embedded by pointer as well. 
I will just update `BaseController` to use value receivers.